crates/gem_hypercore/src/signer/core_signer.rs

```diff
@@ -158,27 +159,72 @@ impl HyperCoreSigner {
         self.sign_serialized_action(delegate, timestamp, private_key, token_delegate_typed_data, "token delegate")
     }
 
-    fn sign_market_message(&self, perpetual_type: &PerpetualType, agent_key: &[u8], builder: Option<Builder>, timestamp: u64) -> SignerResult<String> {
+    fn sign_market_message(
+        &self,
+        perpetual_type: &PerpetualType,
+        agent_key: &[u8],
+        builder: Option<&Builder>,
+        timestamp_incrementer: &mut NumberIncrementer,
+    ) -> SignerResult<Vec<String>> {
         let (data, is_open) = match perpetual_type {
+            PerpetualType::Modify(modify_data) => return self.sign_modify_orders(modify_data, agent_key, builder, timestamp_incrementer),
             PerpetualType::Open(data) | PerpetualType::Increase(data) => (data, true),
             PerpetualType::Close(data) => (data, false),
             PerpetualType::Reduce(reduce_data) => (&reduce_data.data, false),
         };
 
+        let order = Self::market_order_from_confirm_data(data, is_open, builder);
+        Ok(vec![self.sign_place_order(order, timestamp_incrementer.next_val(), agent_key)?])
+    }
+
+    fn sign_modify_orders(
+        &self,
+        modify_data: &PerpetualModifyConfirmData,
+        agent_key: &[u8],
+        builder: Option<&Builder>,
+        timestamp_incrementer: &mut NumberIncrementer,
+    ) -> SignerResult<Vec<String>> {
+        modify_data
+            .modify_types
+            .iter()
+            .map(|modify_type| match modify_type {
+                PerpetualModifyPositionType::Cancel(orders) => {
+                    let cancels = orders.iter().map(|o| CancelOrder::new(o.asset_index as u32, o.order_id)).collect();
+                    self.sign_cancel_order(Cancel::new(cancels), timestamp_incrementer.next_val(), agent_key)
+                }
+                PerpetualModifyPositionType::Tpsl(tpsl) => {
+                    let order = make_position_tp_sl(
+                        modify_data.asset_index as u32,
```

Comment:
```suggestion
                        modify_data.asset_index as u32,
```

can we remove casting